### PR TITLE
Years range selection

### PR DESCRIPTION
**Pull request by DarkerTH**

### CODE REVIEW

**Comment by niumis** [comment]
+1

File: src/AppBundle/Service/Subscription.php [hunk 1]
```diff
@@ -178,4 +199,42 @@ public function setModelId($modelId)
...
+    }
+
+    /**
+     * @param int $yearTo
+     * @return Subscription
+     */
+    public function setYearTo(int $yearTo): Subscription
+    {
+        $this->yearTo = $yearTo;
+        return $this;
```
Comment (zyrukas) [inline src/AppBundle/Service/Subscription.php]: space

File: src/AppBundle/Service/Subscription.php [hunk 2]
```diff
@@ -178,4 +199,42 @@ public function setModelId($modelId)
...
+    }
+
+    /**
+     * @param int $yearFrom
+     * @return Subscription
+     */
+    public function setYearFrom(int $yearFrom): Subscription
+    {
+        $this->yearFrom = $yearFrom;
+        return $this;
```
Comment (zyrukas) [inline src/AppBundle/Service/Subscription.php]: space

File: src/AppBundle/Resources/views/Home/Components/SearchBox/year-list.html.twig [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{% set currentYear = "now"|date("Y") %}
+{% for year in currentYear..1985 %}
+    {% include'@App/Home/Components/SearchBox/year-element.html.twig' with {'year': year, 'id': id} %}
+{% endfor %}
+
+{% for year in 1980..1925 %}
+
```
Comment (zyrukas) [inline src/AppBundle/Resources/views/Home/Components/SearchBox/year-list.html.twig]: really annoying space

File: src/AppBundle/Resources/scripts/main.js [hunk 4]
```diff
@@ -51,19 +65,25 @@ $(document).ready(function () {
...
-        if (brandId <= 0 || modelId <= 0 || isNaN(brandId) || isNaN(modelId)) {
-            alert('Pasirinkite markę ir modelį!');
-
+        if (!validateBrandModel(brandId, modelId) || !validateYearRange(yearFrom, yearTo)){
             return false;
         }
 
+        let yearQuery = '';
+        if (yearFrom > 0 || yearTo > 0){
+            yearQuery = '?yearFrom='+yearFrom+'&yearTo='+yearTo;
```
Comment (zyrukas) [inline src/AppBundle/Resources/scripts/main.js]: psr-2 ¯\_(ツ)_/¯

**Pull request merged**